src/tools/database/elasticsearch.ts

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Tool,
+  ToolInput,
+  ToolError,
+  BaseToolOptions,
+  BaseToolRunOptions,
+  JSONToolOutput,
+  ToolInputValidationError,
+} from "@/tools/base.js";
+import { Cache } from "@/cache/decoratorCache.js";
+import { RunContext } from "@/context.js";
+import { z } from "zod";
+import { ValidationError } from "ajv";
+import { AnyToolSchemaLike } from "@/internals/helpers/schema.js";
+import { Client, ClientOptions } from "@elastic/elasticsearch";
+import {
+  CatIndicesResponse,
+  IndicesGetMappingResponse,
+  SearchRequest,
+  SearchResponse,
+  SearchHit,
+} from "@elastic/elasticsearch/lib/api/types.js";
+
+export interface ElasticSearchToolOptions extends BaseToolOptions {
+  connection: ClientOptions;
+}
+
+export type ElasticSearchToolResult =
+  | CatIndicesResponse
+  | IndicesGetMappingResponse
+  | SearchResponse;
+
+export const ElasticSearchAction = {
+  ListIndices: "LIST_INDICES",
+  getIndexDetails: "GET_INDEX_DETAILS",
+  search: "SEARCH",
+};
+
+export class ElasticSearchTool extends Tool<
+  JSONToolOutput<ElasticSearchToolResult>,
+  ElasticSearchToolOptions,
+  BaseToolRunOptions
+> {
+  name = "ElasticSearchTool";
+
+  description = `Can query data from an ElasticSearch database. IMPORTANT: strictly follow this order of actions:
+   1. ${ElasticSearchAction.ListIndices} - retrieve a list of available indices
+   2. ${ElasticSearchAction.getIndexDetails} - get details of index fields
+   3. ${ElasticSearchAction.search} - perform search or aggregation query on a specific index or pass the original user query without modifications if it's a valid JSON ElasticSearch query`;
+
+  inputSchema() {
+    return z.object({
+      action: z
+        .nativeEnum(ElasticSearchAction)
+        .describe(
+          "The action to perform. LIST_INDICES lists all indices, GET_INDEX_DETAILS fetches details for a specified index, and SEARCH executes a search or aggregation query",
```

Comment:
Retrieve enum values directly instead of hard coding them. 

```
`The action to perform. ${ElasticSearchAction.ListIndices} lists all indices, ${ElasticSearchAction.GetIndexDetails} fetches details for a specified index, and ${ElasticSearchAction.Search} executes a search or aggregation query`
```